### PR TITLE
HSEARCH-4931 Upgrade to Hibernate ORM 6.3.0.Final

### PR DESCRIPTION
**Pull request by marko-bekhta**
https://hibernate.atlassian.net/browse/HSEARCH-4931

### CODE REVIEW

**Comment by marko-bekhta** [comment]
Jenkins, retest this please

**Review by yrodiere — APPROVED** [review]

**Comment by yrodiere** [comment]
@marko-bekhta Thanks. I added a couple commits to follow up on the (implicit) upgrade of byte-buddy.

**Comment by marko-bekhta** [comment]
thanks 😃

**Comment by yrodiere** [comment]
Seems our Nexus cache is broken for some reason. I'll restart it.

**Review by yrodiere — APPROVED** [review]

**Comment by yrodiere** [comment]
@marko-bekhta the build failed, looks like you need to adjust hibernate-search-reports to not depend on Spring ITs

**Comment by marko-bekhta** [comment]
I've moved them into a profile, the same as we have our java-modules tests.

File: build/reports/pom.xml [hunk 1]
```diff
@@ -253,6 +238,30 @@
                 </plugins>
             </build>
         </profile>
+        <profile>
+            <!--
+                We want to keep these module path tests within their own profile,
```
Comment (yrodiere) [inline build/reports/pom.xml:243]:
```suggestion
                We want to keep these Spring tests within their own profile,
```

**Review by yrodiere — CHANGES_REQUESTED** [review]

**Review by yrodiere — APPROVED** [review]

**Comment by marko-bekhta** [comment]
ok 😃 build looks better now I'll merge it and then will rebase the tenancy PR 
thanks!

**Pull request merged**